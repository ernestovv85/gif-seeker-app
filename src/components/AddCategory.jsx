import PropTypes from 'prop-types'
import { useState } from 'react'

export const AddCategory = ({ onNewCategory }) => {
  const [value, setValue] = useState('')

  const onInputChange = ({target}) => {
    setValue(target.value)
  }

  const onSubmit = (event) => {
    event.preventDefault()
    if (value.trim().length <= 0) return
    onNewCategory(value.trim())
    setValue('')
  }

  return (
    <form onSubmit={onSubmit} aria-label='form'>
      <input
        type='text'
        placeholder='Buscar gif'
        value={value}
        onChange={onInputChange}
      />
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
