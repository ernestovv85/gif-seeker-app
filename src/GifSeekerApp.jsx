import { useState } from 'react'
import { AddCategory, GifGrid } from './components'
const GifSeekerApp = () => {
  const [categories, setCategories] = useState([])

  const onAddCategory = (newCategory) => {
    if (categories.includes(newCategory)) return
    setCategories([newCategory, ...categories])
  }
  return (
    <>
      <h1>Gif Seeker App</h1>
      <AddCategory onNewCategory={onAddCategory} />

      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  )
}

export default GifSeekerApp
