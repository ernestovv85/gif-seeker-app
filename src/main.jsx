import React from 'react'
import ReactDOM from 'react-dom/client'
import GifSeekerApp from './GifSeekerApp.jsx'
import './styles.css'

ReactDOM.createRoot(document.getElementById('root')).render(
  <React.StrictMode>
    <GifSeekerApp />
  </React.StrictMode>,
)
