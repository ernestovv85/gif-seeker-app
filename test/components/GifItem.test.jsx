/* eslint-disable no-undef */
import { render, screen } from '@testing-library/react'
import { GifItem } from '../../src/components'
describe('Pruebas en <GifItem />', () => {
  const title = 'one punch'
  const url = 'https://onepunch.com/saitama.jpg'

  test('debe hacer match con el snapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  })

  test('debe mostrar la imagen con url indicado y alt indicado', () => {
    render(<GifItem title={title} url={url} />)
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  })

  test('debe mostrar el tÃ­tulo en el componente', () => { 
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
   })
})
