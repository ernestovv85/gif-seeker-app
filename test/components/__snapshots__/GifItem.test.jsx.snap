// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="one punch"
      src="https://onepunch.com/saitama.jpg"
    />
    <p>
      one punch
    </p>
  </div>
</div>
`;
